_delphi_utils_python/delphi_utils/covidcast_wrapper.py

```diff
@@ -0,0 +1,251 @@
+"""module for covidcast api call wrapper."""
+
+from datetime import date, timedelta
+from typing import Iterable, Union
+
+import pandas as pd
+from delphi_epidata import Epidata
+from epiweeks import Week
+
+
+def date_generator(startdate: date, enddate: date) -> Iterable[date]:
+    """
+    Take start date and end date and generates date string.
+
+    Parameters
+    ----------
+    startdate: date
+    enddate: date
+
+    Returns
+    -------
+    generator of str
+    """
+    while startdate <= enddate:
+        yield startdate.strftime("%Y-%m-%d")
+        startdate = startdate + timedelta(days=1)
+
+
+def _parse_datetimes(date_int: int, time_type: str, date_format: str = "%Y%m%d") -> Union[pd.Timestamp, None]:
+    """Convert a date or epiweeks string into timestamp objects.
+
+    Datetimes (length 8) are converted to their corresponding date, while epiweeks (length 6)
+    are converted to the date of the start of the week. Returns nan otherwise
+
+    Epiweeks use the CDC format.
+
+    :param date_int: Int representation of date.
+    :param date_format: String of the date format to parse.
+    :returns: Timestamp.
+    """
+    date_str = str(date_int)
+    if time_type == "day":
+        return pd.to_datetime(date_str, format=date_format)
+    if time_type == "week":
+        epiwk = Week(int(date_str[:4]), int(date_str[-2:]))
+        return pd.to_datetime(epiwk.startdate())
+    return None
+
+
+def metadata() -> Union[pd.DataFrame, None]:
+    """
+    Make covidcast metadata api call.
+
+    Returns
+    -------
+    pd.DataFrame of covidcast metadata.
+    """
+    # pylint: disable=W0212
+    response = Epidata._request("covidcast_meta")
+
+    if response["result"] != 1:
+        # Something failed in the API and we did not get real metadata
+        raise RuntimeError("Error when fetching metadata from the API", response["message"])
+
+    df = pd.DataFrame.from_dict(response["epidata"])
+    df["min_time"] = df.apply(lambda x: _parse_datetimes(x.min_time, x.time_type), axis=1)
+    df["max_time"] = df.apply(lambda x: _parse_datetimes(x.max_time, x.time_type), axis=1)
+    df["last_update"] = pd.to_datetime(df["last_update"], unit="s")
+    return df
+
+
+def signal(
+    data_source: str,
+    signal: str,  # pylint: disable=W0621
+    start_day: date = None,
+    end_day: date = None,
+    geo_type: str = "county",
+    geo_values: Union[str, Iterable[str]] = "*",
+    as_of: date = None,
+    lag: int = None,
+    time_type: str = "day",
+) -> Union[pd.DataFrame, None]:
+    """Download a Pandas data frame for one signal.
```

Comment:
This docstring looks like it was copied from https://github.com/cmu-delphi/covidcast/blob/main/Python-packages/covidcast-py/covidcast/covidcast.py ...  I dont think we need this much detail here.  We arent necessarily trying to duplicate the exact functionality of the `covidcast` client, we just wanna replace it with something else that does a similar job but with a simpler interface and less dependency bloat.